app/routes/_gcn.circulars/circulars.server.ts

```diff
@@ -301,6 +301,36 @@ export async function circularRedirect(query: string) {
   }
 }
 
+export async function putVersion(
+  circular: Omit<Circular, 'createdOn' | 'submitter' | 'submittedHow'>,
+  user?: User
+): Promise<number> {
+  validateCircular(circular.subject, circular.body)
+  if (!user?.groups.includes(moderatorGroup))
+    throw new Response('User is not a moderator', {
+      status: 403,
+    })
+  const circularVersionsAutoIncrement = await getDynamoDBVersionAutoIncrement(
+    circular.circularId
+  )
+
+  // Need to be retrieved otherwise will be absent in latest version
+  const { bibcode, submitter, submittedHow } = await get(circular.circularId)
+
+  const newCircularVersion = {
+    ...circular,
+    editedBy: formatAuthor(user),
+    createdOn: Date.now(),
+    bibcode,
+    submitter,
+    submittedHow,
+  }
+  // Remove fields if they are undefined, as this will otherwise throw an error in the PUT
+  if (!bibcode) delete newCircularVersion.bibcode
+  if (!submittedHow) delete newCircularVersion.submittedHow
```

Comment:
If we use spread syntax here, then we don't need to manually delete undefined properties:

```suggestion
  // Need to be retrieved otherwise will be absent in latest version
  const oldCircular = await get(circular.circularId)

  const newCircularVersion = {
    ...oldCircular,
    ...circular,
    editedBy: formatAuthor(user),
    createdOn: Date.now(),
  }
  if (!submittedHow) delete newCircularVersion.submittedHow
```

But... uh oh, we probably _don't_ want to modify the createdOn field. Perhaps we need a new revisedOn field. (That should be a separate PR since it affects backend code that is already on the main branch.)

Reply:
Added https://github.com/nasa-gcn/gcn.nasa.gov/pull/1802 and I will implement the spread syntax next